### PR TITLE
docs: Add visualization tool usage in readme

### PR DESCRIPTION
**Pull request by yahia3200**
## Description

Visualization tool usage documentation

## Changes Made
- Update the readme file to mention the visualization tool 
- Add a usage file to describe how to use the tool

### CODE REVIEW

File: misc/visualization_tool_images/.gif
Comment (ZetaTwo) [inline misc/visualization_tool_images/.gif:1]: What is this file?

**Review by ZetaTwo — CHANGES_REQUESTED** [review]
Could you try to see if you can make the animations .webp files instead of .gif to improve both file size and image quality?

**Comment by yahia3200** [comment]
> Could you try to see if you can make the animations .webp files instead of .gif to improve both file size and image quality?

OK, I will try to

File: misc/visualization_tool_images/.gif
Comment (yahia3200) [reply misc/visualization_tool_images/.gif:1]: It is a wrong file, I will delete it

Comment (yahia3200) [reply misc/visualization_tool_images/.gif:1]: Done

**Comment by yahia3200** [comment]
I found that GitHub can't render webp images 
https://docs.github.com/en/repositories/working-with-files/using-files/working-with-non-code-files
Do you suggest to try find some hacks around it or leave it for now?

**Comment by ZetaTwo** [comment]
That list is not exhaustive, see https://github.com/TomasHubelbauer/markdown-webp for a working example (code: https://github.com/TomasHubelbauer/markdown-webp/blob/main/readme.md?plain=1) so it really should work.

**Comment by yahia3200** [comment]
It wasn't working on my vscode preview which made me search if it was supported. I tried again and it turns out that I only need to refresh the preview 😅​

**Review by ZetaTwo — APPROVED** [review]

**Comment by ZetaTwo** [comment]
Really nice work!

**Pull request merged**